components/style/dom_apis.rs

```diff
@@ -276,15 +277,15 @@ where
     }
 
     if root.is_in_document() {
-        return root.owner_doc().elements_with_id(id);
-    }
+        if let Some(shadow) = root.as_shadow_root() {
```

Comment:
This is still wrong and doesn't seem addressed. is_in_document() and as_shadow_root().is_some() are mutually exclusive. This should be reverted.